src/components/standalone/certificates/CreateLetsEncryptCertificateDrawer.vue

```diff
@@ -0,0 +1,327 @@
+    dnsApiOptionsValueValidationErrors.value.push('')
+  })
+
+  let validDomains = true
+  for (let [index, domain] of domains.value.entries()) {
+    if (domain) {
+      let validator = validateFQDN(domain)
+      if (!validator.valid) {
+        domainValidationErrors.value[index] = t(validator.errMessage as string)
+        validDomains = false
+        break
+      }
+    } else {
+      domainValidationErrors.value[index] = t('error.required')
+      validDomains = false
+    }
+  }
+
+  let validDnsOptions = true
+
+  if (validationMethod.value === 'dns') {
+    validDnsOptions = runValidators([validateRequired(dnsApi.value)], 'dns_provider')
+
+    for (let [index, dnsApiOption] of dnsApiOptions.value.entries()) {
+      if (!dnsApiOption.key) {
+        dnsApiOptionsKeyValidationErrors.value[index] = t('error.required')
+        validDnsOptions = false
+      }
+      if (!dnsApiOption.value) {
+        dnsApiOptionsValueValidationErrors.value[index] = t('error.required')
+        validDnsOptions = false
+      }
+    }
+  }
+
+  return (
+    validDomains &&
+    runValidators(
+      [validateRequired(certificateName.value), validateUciName(certificateName.value)],
+      'name'
+    ) &&
+    validDnsOptions
+  )
+}
+
+async function createCertificate() {
+  if (!validate()) {
+    return
+  }
+
+  try {
+    isSavingChanges.value = true
+    await ubusCall('ns.reverseproxy', 'request-certificate', {
+      name: certificateName.value,
+      domains: domains.value,
+      validation_method: validationMethod.value,
+      ...(validationMethod.value === 'dns'
+        ? {
+            dns_provider: dnsApi.value,
+            dns_provider_options: dnsApiOptions.value.map((x) => `${x.key}=${x.value}`)
+          }
+        : {})
+    })
+    isSavingChanges.value = false
+    emit('add-certificate')
+    close()
+  } catch (err: any) {
+    if (err instanceof ValidationError) {
+      validationErrorBag.value = err.errorBag
+    } else {
+      error.value.notificationTitle = t('error.cannot_create_certificate')
+      error.value.notificationDescription = t(getAxiosErrorMessage(err))
+    }
+    isSavingChanges.value = false
+  }
+}
+
+watch(
+  () => props.isShown,
+  () => {
+    if (props.isShown) {
+      fetchOptions().then(() => {
+        resetForm()
+      })
+    }
+  }
+)
+</script>
+
+<template>
+  <NeSideDrawer
+    :is-shown="isShown"
+    @close="close()"
+    :closeAriaLabel="t('standalone.shell.close_side_drawer')"
+    :title="t('standalone.certificates.add_lets_encrypt_certificate')"
+  >
+    <NeInlineNotification
+      v-if="error.notificationTitle"
+      :title="error.notificationTitle"
+      :description="error.notificationDescription"
+      class="mb-6"
+      kind="error"
+      ><template v-if="error.notificationDetails" #details>
+        {{ error.notificationDetails }}
+      </template></NeInlineNotification
+    >
+    <NeSkeleton :lines="20" v-if="loading" />
```

Comment:
Didn't see it in action, but 20 lines seem a lot :) Consider showing fewer lines, e.g. 10 and using size='lg' for the skeleton